pkg/httputil/client_trace.go

```diff
@@ -0,0 +1,27 @@
+package httputil
+
+import (
+	"context"
+	"net/http/httptrace"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var connectionGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+	Name: "out_conns",
+	Help: "A gauge of in-use TCP connections",
+}, []string{"service"})
+
+func SetClientTrace(ctx context.Context, service string) context.Context {
+	trace := &httptrace.ClientTrace{
+		GotConn: func(info httptrace.GotConnInfo) {
+			connectionGauge.WithLabelValues(service).Inc()
+		},
+		PutIdleConn: func(err error) {
+			connectionGauge.WithLabelValues(service).Dec()
+		},
+	}
```

Comment:
I am not sure how this implements a "metric for open connections".  It seems like it implements a "metric for connections in use", which is different.  I am happy with either way - but let's not call the metric `out_conns`, and ideally explain this in the PR title and description

Reply:
I think again you might have intended to push another commit.